vendor/k8s.io/kube-openapi/pkg/aggregator/aggregator.go

```diff
@@ -0,0 +1,276 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregator
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/go-openapi/spec"
+
+	"k8s.io/kube-openapi/pkg/util"
+)
+
+const (
+	definitionPrefix = "#/definitions/"
+)
+
+// Run a walkRefCallback method on all references of an OpenAPI spec
+type referenceWalker struct {
+	// walkRefCallback will be called on each reference and the return value
+	// will replace that reference. This will allow the callers to change
+	// all/some references of an spec (e.g. useful in renaming definitions).
+	walkRefCallback func(ref spec.Ref) spec.Ref
+
+	// The spec to walk through.
+	root *spec.Swagger
+}
+
+func walkOnAllReferences(walkRef func(ref spec.Ref) spec.Ref, sp *spec.Swagger) {
+	walker := &referenceWalker{walkRefCallback: walkRef, root: sp}
+	walker.Start()
+}
+
+func (s *referenceWalker) walkRef(ref spec.Ref) spec.Ref {
+	if ref.String() != "" {
+		refStr := ref.String()
+		// References that start with #/definitions/ has a definition
+		// inside the same spec file. If that is the case, walk through
+		// those definitions too.
+		// We do not support external references yet.
+		if strings.HasPrefix(refStr, definitionPrefix) {
```

Comment:
`strings.HasPrefix(refStr, definitionPrefix)` implies `ref.String() != ""`. No need for the first if-clause at all.